src/dmd/root/filename.d

```diff
@@ -631,71 +631,71 @@ nothrow:
         }
     }
 
+    /**
+       Ensure that the provided path exists
+
+       Accepts a path to either a file or a directory.
+       In the former case, the basepath (path to the containing directory)
+       will be checked for existence, and created if it does not exists.
+       In the later case, the directory pointed to will be checked for existence
+       and created if needed.
+
+       Params:
+         path = a path to a file or a directory
+
+       Returns:
+         `true` if the directory exists or was successfully created
+     */
     extern (C++) static bool ensurePathExists(const(char)* path)
     {
         //printf("FileName::ensurePathExists(%s)\n", path ? path : "");
-        if (path && *path)
+        if (!path || !(*path))
+            return true;
+        if (exists(path))
+            return true;
+
+        // We were provided with a file name
+        // We need to call ourselves recursively to ensure parent dir exist
+        const(char)* p = FileName.path(path);
+        if (*p)
         {
-            if (!exists(path))
+            version (Windows)
             {
-                const(char)* p = FileName.path(path);
-                if (*p)
+                const len = strlen(path);
+                if ((len > 2 && p[-1] == ':' && path[2 .. len] == p[0 .. strlen(p)])
+                     || len == strlen(p))
```

Comment:
Note that `strlen(p)` appears twice. The `p[-1]` is a buffer overflow bug because `Filename.path()` does not guarantee there is anything there.